### PR TITLE
chore: upgrade to beta-4 dependencies

### LINKED ISSUE #1167: Upgrade dependencies for beta-4
- Blocked by https://github.com/FuelLabs/fuel-indexer/issues/1074
- fuel-core 0.20.x
- fuels-rs (latest version that's compatible with fuel-core 0.20)
- We should not merge this until the beta-4 network is released

### PR DESCRIPTION
**Pull request by deekerno**
Closes #1167.

### Description

Upgrades the indexer to beta-4-compatible dependencies.

### Testing steps

CI should pass.

As a manual test, run the explorer example and make sure that queries work properly.

### CODE REVIEW

**Comment by deekerno** [comment]
This PR is currently blocked until `fuels-rs` is released with support for `fuel-core v0.20.1`; the SDK team has stated that it will happen very soon.

File: .github/CODEOWNERS
Comment (ra0x3) [inline .github/CODEOWNERS]: Was this change errantly included?

File: Cargo.toml
Comment (ra0x3) [inline Cargo.toml:63]: 👌🏽

File: packages/fuel-indexer-tests/src/fixtures.rs
Comment (ra0x3) [inline packages/fuel-indexer-tests/src/fixtures.rs:529]: Why are these tests flaky?

File: plugins/forc-index/src/defaults.rs
Comment (ra0x3) [inline plugins/forc-index/src/defaults.rs]: Any reason this is different than the rust-version specified in the workspace manifest? (1.71.1 vs 1.71.0)

**Review by ra0x3 — COMMENTED** [review]
@deekerno 
- Looks mostly ok thanks for getting to this
- Left some questions
- Also don't forget to update the rustc version in `ci.yml`

**Comment by deekerno** [comment]
> @deekerno 
> 
> - Looks mostly ok thanks for getting to this
> 
> - Left some questions
> 
> - Also don't forget to update the rustc version in `ci.yml`

@ra0x3 Looks like a rebase I did when moving computers blew away some of the changes I had 😅 will fix and re-request.

File: packages/fuel-indexer-tests/src/fixtures.rs
Comment (deekerno) [reply packages/fuel-indexer-tests/src/fixtures.rs:529]: There seems to have been a change made between `v0.40` and `v0.45` regarding how to get variable outputs, which are necessary for the `*Out` receipt tests. I'm already making changes to the E2E tests as part of #1194, so I'll fix and re-enable them as part of that work.

**Review by ra0x3 — APPROVED** [review]
- Code looks fine, thanks @deekerno 
- Merging this into the feature branch so as to not block anything
- We will wait until beta-4.fuel.network is up to test this stuff
- After we manually test that, we can merge

**Pull request merged**
